File: src/unifyweaver/targets/python_runtime/llm.py
import subprocess
import sys
import shutil

class LLMProvider:
    def __init__(self, model="gemini-2.0-flash-exp"):
        self.model = model
        self.cli_path = shutil.which("gemini")

    def ask(self, prompt, context=None):
        if not self.cli_path:
            return "Error: gemini CLI not found in PATH"

        full_prompt = prompt
        if context:
            full_prompt += "\n\nContext:\n" + str(context)
            
        cmd = [self.cli_path, "-p", full_prompt, "--model", self.model]
        
        try:
            # Run CLI (capture output)
            result = subprocess.run(cmd, capture_output=True, text=True)
            if result.returncode != 0:
                return f"Error running gemini: {result.stderr}"
            return result.stdout.strip()
        except Exception as e:
            return f"Exception calling LLM: {e}"

File: examples/pearltrees/llm.rs
use std::process::Command;
use serde_json::Value;

pub struct LLMProvider {
    model: String,
}

impl LLMProvider {
    pub fn new(model: &str) -> Self {
        Self {
            model: model.to_string(),
        }
    }

    pub fn ask(&self, prompt: &str, context: &Vec<Value>) -> Result<String, Box<dyn std::error::Error>> {
        let context_str = serde_json::to_string_pretty(context)?;
        let full_prompt = format!("{}\n\nContext:\n{}", prompt, context_str);

        // Call gemini CLI
        let output = Command::new("gemini")
            .arg("-p")
            .arg(&full_prompt)
            .arg("--model")
            .arg(&self.model)
            .output()?;

        if output.status.success() {
            Ok(String::from_utf8_lossy(&output.stdout).to_string())
        } else {
            let err = String::from_utf8_lossy(&output.stderr);
            Err(format!("LLM Error: {}", err).into())
        }
    }
}

